### ISSUE TITLE
Fix S3263 FP: Static initialization is not detected in struct

### ISSUE THREAD

**Issue by Corniel** [SonarSource/sonar-dotnet#3187]
### Description
Static fields should appear in the order they must be initialized, but for array (and list) initialization, this is not detected correctly. Note, the noncompliant is already detected correctly, but might be a good test case when changing the behaviour of the rule.

### Repro steps
``` C#
public struct S3263
{
    public static readonly S3263 Zero;
    public static readonly S3263 V42 = new S3263(42);

    public static readonly IReadOnlyCollection<S3263> All = new[] { Zero, V42, }; // compliant, both values have been initialized.

    public static readonly IReadOnlyCollection<S3263> Extra = new[] { Zero, V007, V42, }; // noncompliant, V007 has not been initalized.

    public static readonly S3263 V007 = new S3263(7);

    public S3263(int i) => Value = i;

    public int Value { get; }
}
```

### Related information
* SonarC# 8.4

**Comment by pavel-mikula-sonarsource** [comment]
Hello @Corniel ,
arrays are handled as expected, problem is that the rule does not work on struct.
